src/types/rules.ts

```diff
@@ -0,0 +1,12 @@
+import { Commit } from "@commitlint/parse";
+
+export type RuleCheckResults = null | Record<string, unknown>;
```

Comment:
These types can be moved to `lint` module. We'd like to localized the types to where it is mostly coupled.